chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -571,23 +587,43 @@ func (b *BitcoindNotifier) handleBlockConnected(block chainntnfs.BlockEpoch) err
 		return fmt.Errorf("unable to get block: %v", err)
 	}
 
-	chainntnfs.Log.Infof("New block: height=%v, sha=%v",
-		block.Height, block.Hash)
-
 	txns := btcutil.NewBlock(rawBlock).Transactions()
 	err = b.txConfNotifier.ConnectTip(
 		block.Hash, uint32(block.Height), txns)
 	if err != nil {
 		return fmt.Errorf("unable to connect tip: %v", err)
 	}
 
+	chainntnfs.Log.Infof("New block: height=%v, sha=%v", block.Height,
+		block.Hash)
+
 	// We want to set the best block before dispatching notifications so
 	// if any subscribers make queries based on their received block epoch,
 	// our state is fully updated in time.
 	b.bestBlock = block
 
 	b.notifyBlockEpochs(block.Height, block.Hash)
 
+	// Finally, we'll update the spend height hint for all of our watched
+	// outpoints that have not been spent yet. This is safe to do as we do
+	// not watch already spent outpoints for spend notifications.
+	ops := make([]wire.OutPoint, 0, len(b.spendNotifications))
+	for op := range b.spendNotifications {
+		ops = append(ops, op)
+	}
+
+	if len(ops) > 0 {
+		err := b.spendHintCache.CommitSpendHint(
```

Comment:
Not sure if the ordering here is crucial. The reason I made it to commit the new height hint after notifying is because we don't keep track of things that have already been notified, so we can just simply iterate over the current pending notifications.

Reply:
It seems like this call is actually in the wrong place then, we only dispatch spend ntfns when they come through as relevant transaction. I'm not sure what the ordering guarantees are on those vs delivery of blocks, but it seems like it could be a little racey

Reply:
Not sure if there's a good way around this though, as we need to update the spend hints at some point.

Reply:
I think we can probably merge it as is and then revisit later if we think of a more concrete solution